oak_restricted_kernel_sdk/src/dice.rs

```diff
@@ -15,15 +15,67 @@
 //
 
 use anyhow::Ok;
-use oak_crypto::encryptor::{EncryptionKeyProvider, RecipientContextGenerator};
+use oak_crypto::encryptor::EncryptionKeyProvider;
 use oak_dice::evidence::{Evidence, RestrictedKernelDiceData, P256_PRIVATE_KEY_SIZE};
 use oak_restricted_kernel_interface::{syscall::read, DICE_DATA_FD};
 use p256::ecdsa::SigningKey;
 use zerocopy::{AsBytes, FromZeroes};
 
+pub use oak_crypto::encryptor::RecipientContextGenerator as EncryptionKeyHandle;
+pub trait Signer {
+    fn sign(&self, message: &[u8]) -> anyhow::Result<oak_crypto::signer::Signature>;
+}
+pub trait Attester {
```

Comment:
Not sure about this naming -- is the definition of an attester just anything that can return evidence?

Reply:
Copied the naming from the design doc, but agreed it isn't ideal. 

> is the definition of an attester just anything that can return evidence?

yes.

`Evidence` may be a more descriptive term. cc @ipetr0v

Reply:
Personally I'm okay with the name, but one could argue that it doesn't quite meet the definition outlined in the RFC 

> The Attester role is assigned to entities that create Evidence that is conveyed to a Verifier.

this struct just retrieves evidence, it doesn't do any attesting.

Reply:
> this struct just retrieves evidence, it doesn't do any attesting.

This is just the SDK interface, and the creation is hidden behind it (even though the `Attester` itself just retrieves it). So from the point of view of an app developer - it creates an Evidence

Reply:
> So from the point of view of an app developer - it creates an Evidence

Disagree, even from the point of view of the app developer, the evidence precedes this struct, it merely exposes it. 

An developer may create an `InstanceSigner` first. Yet the private key it uses for signing is implicitly linked to the public key in the `Evidence` struct created later.